File: official/docs/csharp/current/carbon-offset/one-call-buy.cs
using EasyPost;
EasyPost.ClientManager.SetCurrent("<YOUR_TEST/PRODUCTION_API_KEY>");

Dictionary<string, object> fromAddress = new Dictionary<string, object>() {
  { "name", "Dr. Steve Brule" },
  { "street1", "417 Montgomery Street" },
  { "street2", "5th Floor" },
  { "city", "San Francisco" },
  { "state", "CA" },
  { "country", "US" },
  { "zip", "94104" }
};

Dictionary<string, object> toAddress = new Dictionary<string, object>() {
  { "company", "EasyPost" },
  { "street1", "417 Montgomery Street" },
  { "street2", "Floor 5" },
  { "city", "San Francisco" },
  { "state", "CA" },
  { "country", "US" },
  { "zip", "94104" }
};

Shipment shipment = await Shipment.Create(new Dictionary<string, object>() {
  { "carrier_accounts", new Dictionary<string, string>() { { “id”, “ca_...” } } },
  { "service", “NextDayAir” }
  { "parcel", new Dictionary<string, object>() {
    { "length", 8 }, { "width", 6 }, { "height", 5 }, { "weight", 10 }
  } },
  { "to_address", toAddress },
  { "from_address", fromAddress },
  { "reference", "ShipmentRef" }
}, withCarbonOffset: true);

File: official/docs/golang/current/carbon-offset/one-call-buy.go
package example

import (
    "os"

    "github.com/EasyPost/easypost-go/v2"
)

func main() {
    apiKey := os.Getenv("<YOUR_TEST/PRODUCTION_API_KEY>")
    client := easypost.New(apiKey)

    toAddress, err := client.GetAddress("adr_...")
    fromAddress, err := client.GetAddress("adr_...")
    parcel, err := client.GetParcel("prcl_...")

    shipment, err := client.CreateShipmentWithCarbonOffset(
        &easypost.Shipment{
            CarrierAccountIDs: []string{"ca_..."},
            Service:           "NextDayAir",
            Parcel:            parcel,
            ToAddress:         toAddress,
            FromAddress:       fromAddress,
            Reference:         "ShipmentRef",
        },
    )
}
